tools/shared.py

```diff
@@ -1141,13 +1135,10 @@ def __setattr__(self, attr, value):
           self.attrs.pop(a, None)
 
       if attr in self.legacy_settings:
-        if self.attrs['STRICT']:
```

Comment:
why disable strict checks here?

Reply:
The warning manager now handles this.  So rather than controlling it via `-s STRICT` you can use `-Werror` or `-Wlegacy-settings`.

I think this works especially well if we have `-s STRICT` enable `WARNINGS_ARE_ERRORS` so we get the best of both worlds.